src/wpilib/i2c.rs

```diff
@@ -0,0 +1,134 @@
+use wpilib::wpilib_hal::*;
+use wpilib::usage::*;
+use wpilib::hal_call::*;
+
+use std::slice;
+
+/// Helper enum that describes one of two possible locations on the roborio for i2c devices
+#[derive(Copy, Clone)]
+pub enum I2cPort {
+    /// Enum option for i2c device plugged into the onboard i2c connectors on the roborio
+    OnBoard = 0,
+    /// Enum option for i2c device, such as a navX board, connected through the MXP port
+    MXP = 1,
+}
+
+/// Describes various errored or otherwise special cases that may result from an i2c transaction
+pub enum I2cError {
+    // TODO find is there is any way to get more i2c transaction information out of HAl than just
+    // success or failure
+
+    /// Indicates a general error in an i2c transaction
+    IOError,
+    /// Indicates when no string, or an invalid string was received from the i2c transaction
+    InvalidReceiveString,
+    /// Indicates a situation where the connection was specifically aborted
+    TransferAbort,
+}
+
+/// Struct for sending and receiving data over i2c
+pub struct I2cInterface {
+    port: I2cPort,
+    device_address: i32,
+}
+
+impl I2cInterface {
+    /// users create a new I2cInterface here
```

Comment:
Just for consistency, it would be good if documentation comments were properly capitalized and in the imperative tense ("Create a new I2cInterface.")